### PR TITLE
Weather app-Elle Frankenberg and Siri Hemsley

### PR DESCRIPTION
**Pull request by ElleFrankenberg**
Link to my deployed project:
https://loving-feynman-2c374e.netlify.app

My reflections on how this week's project turned out:
It was really fun to learn about how to use API:s, and it was good to have some designs to use and try to mimic. Over all we are very happy with the result. And that we did almost the red goals.

Things I'd like to have clarified or explained in more detail:
It would be good to get more explanation about how to get items in arrays, since that was something we where struggleing with.

### CODE REVIEW

File: Code/script.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+fetch(currentWeatherUrl)
+ .then((response) => {
+  return response.json()
+ })
+ .then((data) => {
+   const weatherDescription = data.weather.map((value) => {
+    return value.main
+    })
+  
+    currentWeather.innerHTML = `<p>${weatherDescription} | ${data.main.temp.toFixed()}</p>`
```
Comment (idasylven) [inline Code/script.js:19]: You have missed the Celsius sign for the current temperature 🌡️

File: Code/script.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    currentWeather.innerHTML += `<p>sunrise ${new Date(data.sys.sunrise * 1000).toLocaleString('se-SE', {hour:'numeric', minute: 'numeric'})}</p>`
+    currentWeather.innerHTML += `<p>sunset ${new Date(data.sys.sunset * 1000).toLocaleString('se-SE', {hour:'numeric', minute: 'numeric'})}</p>`
+
+  
+  //mood.innerHTML += `<h2>Light a fire and get cosy. ${data.name} is looking grey today.</h2>`
```
Comment (idasylven) [inline Code/script.js:24]: Remove commented out code?

File: Code/script.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  const weatherMessage = () => {
+    if (weatherDescription[0] === "Clouds") {
+      console.log('working')
```
Comment (idasylven) [inline Code/script.js:27]: Remove console.log

File: Code/script.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  fetch(forecastUrl)
+  .then((response) => {
+    return response.json()
+   })
+   .then((data) => {
+    const filteredForecast = data.list.filter(item => item.dt_txt.includes('12:00'))
+
+    filteredForecast.forEach((value) => {
+      const forecastDate = new Date(value.dt * 1000);
+      console.log(value.dt)
```
Comment (idasylven) [inline Code/script.js:61]: Remove console.log

File: Code/script.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      mood.innerHTML += `<img src="icons/Umbrella.svg">`
+      mood.innerHTML += `<h2>Don't forget your umbrella! It's wet in ${data.name} today.</h2>`
+      body.classList.add("rain")
+    } else if (weatherDescription[0] === "Sun") {
+      mood.innerHTML += `<img src="icons/sun.svg">`
+      mood.innerHTML += `<h2>Get sunnies on! ${data.name} is looking rather great today.</h2>`
+      body.classList.add("sun")
+    } else if (weatherDescription[0] === "Snow") {
+      mood.innerHTML += `<img src="icons/snowflake.svg">`
+      mood.innerHTML += `<h2>Put our warmest coat on! ${data.name} is rather chilly today.</h2>`
```
Comment (idasylven) [inline Code/script.js:41]: Is this perhaps a typo? Should "Put our warmest coat on" be "Put your warmest coat on"?

**Review by idasylven — APPROVED** [review]
First of all, the mimicking of provided design option number 2 is outstanding! Very well done, the designer of this layout would be so happy - it´s really on point! 🎨

The responsiveness for mobile, tablet and desktop view works great! All other Blue Level requirements were also met. Nice work making the color of the page change as well as the weather icon depending on the weather/temperature (both parts of Red Level stretch goals) 🌃 🌆

We had no constructive remarks to add in the HTML or CSS section, and only a few small comments regarding the JavaScript. We were very impressed you managed to keep the code so short (especially in the HTML) 👍

Suggestions for improvement:
- Maybe add some explanatory comments for the functions in JS?
- It would be nice with more weather description alternatives in the JS. When we did the peer review the weather was mist - so unfortunately no weather icon was displayed 😞 

Overall the code is very clear and understandably written, and all general guidelines for how to write code attending the Technigo Boot Camp have been followed ☑️

It really would be great to create forecasts for additional cities (just like you mention), if you do get the time. It´s a fantastic app! 🌟

Cheers/
Ida & Anna 🤗

File: Code/script.js (hunk 4 above)
Comment (KarolCislowski) [reply Code/script.js:61]: Yes, remember to always clean your code from all console.log()s and commented out code blocks and tips left for you by teachers 🗡️ ✂️

File: Code/style.css [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+.sun {
+  color: #2A5510;
+  background-color: #F7E9B9;
+}
+
+.snow {
+  color: #000000;
+  background-color: #ffffff;
+}
```
Comment (KarolCislowski) [inline Code/style.css:24]: nice that you made an effort to create different styles for different weather conditions 👍🏻

**Review by KarolCislowski — APPROVED** [review]
This is a very nice and clean project 🌈 
Please remember to always remove all logs and other elements that are helpful during coding but are not welcome in a finished project. 🗑️ 
Great Job!

**Pull request closed without merging**